### DISCUSSION TITLE
No Punctuation

### DISCUSSION THREAD

**Discussion by Avirhel** [openai/whisper#557] [Q&A]
Hello.
Many times stop adding punctuation.
Sometimes start well:
![image](https://user-images.githubusercontent.com/62714086/202855581-7d537c55-6fbd-443b-a8d1-630dc9e999e9.png)
Then stop adding the punctuation:
![image](https://user-images.githubusercontent.com/62714086/202855633-af08abd5-788b-48b5-9894-49095035584f.png)
I'm using:  --model medium.en --initial_prompt "Hello, welcome to my lecture."
Is there a way to somehow force punctuation to always be present?
Or how can I config in a way that I can have better results?
This is awesome. But this is disappointing. 
Thank you.

**Comment by DoubleF3lix** [comment]
Did you ever find an answer to this? I'm also having this issue.

**Reply by glangford** [reply]
Hard to say for certain, but worth trying without `--initial_prompt`

**Reply by DD3Boh** [reply]
That's usually what gets it to add punctuation in the first place, actually.
For example in some of my testings, the only model that got punctuation started by default without an initial prompt is the large-v3 model, all other models didn't add it without adding an initial prompt.

**Comment by Purfview** [comment]
Check `Standalone Faster-Whisper` from there -> https://github.com/Purfview/whisper-standalone-win
It has several experimental options to solve this issue.

**Comment by T145** [comment]
Try using the `large-v2` model, set the language to English, and have your prompt be something more like `"It's just my unique approach to doing it using all the information. I'm not applying every possible scenario."` Also, be sure to clear your GPU cache before processing:
```python
def clear_gpu_cache():
    if torch.cuda.is_available():
        torch.cuda.empty_cache()
```

**Reply by DD3Boh** [reply]
I tried with the `large-v3` model, and I got punctuation there pretty reliably with no initial input as well. The issue is with smaller models, specifically in my case:
- `medium` tends to put the punctuation but only with an initial input
- smaller models don't show it at all or show it unreliably even with an initial input

**Reply by T145** [reply]
Nice! I'm recommending `large-v2` b/c of this comment: https://github.com/openai/whisper/discussions/1884#discussioncomment-7804233

Also it may be good to mark answer so other people can find a solution. 😉

**Comment by sjtu-hxj** [comment]
You can try this: `--initial_prompt "Please do not forget the punctuations!"`

**Reply by sjtu-hxj** [reply]
it worked for my case with `large-v3`

**Comment by leoauri** [comment]
It might be luck, but adding the option `--patience 2.0` just got my transcript to stay with punctuation longer.
Without:
```
[08:32.920 --> 08:36.320]  I mean, I think it's definitely learnable.
[08:36.320 --> 08:37.700]  In terms of like the...
[08:37.700 --> 08:45.300]  virtuosity yeah i mean within uh within the parameters of the sounds that are there yeah
[08:45.300 --> 08:52.420]  i think it's possible again to develop a process and then to be able with practice once one learns
```
With:
```
[08:33.740 --> 08:34.060]  Yeah.
[08:34.220 --> 08:39.700]  I mean, I think it's definitely learnable in terms of, like, the virtuosity.
[08:39.940 --> 08:40.140]  Yeah.
[08:40.280 --> 08:44.100]  I mean, within, uh, within the parameters of the sounds.
```
The complete command I am using for now:
`whisper --language English --model large-v3 --patience 2.0 --initial_prompt "We use all the standard punctuation and capitalization rules of the English language. Sentences start with a capital letter, and end with a full stop. Of course, where appropriate, commas are included. The pronoun \"I\" should be capitalized! So" "$file"`
